src/main/java/com/stripe/model/BalanceTransaction.java

```diff
@@ -146,7 +148,16 @@ public String getType() {
 	public void setType(String type) {
 		this.type = type;
 	}
+	
+	public void setSourceObject(HasId sourceObject) {
+		this.sourceObject = sourceObject;
+	}
 
+	@SuppressWarnings("unchecked")
+	public <O> O getSourceObject() {
+		return (O) sourceObject;
+	}
```

Comment:
@kion We've fairly recently introduced a special type for expandable fields like this that I think we can potentially reuse here. What do you think about this change? (Should apply cleanly to your branch.)

``` patch
diff --git a/src/main/java/com/stripe/model/BalanceTransaction.java b/src/main/java/com/stripe/model/BalanceTransaction.java
index bd86330..79a504f 100644
--- a/src/main/java/com/stripe/model/BalanceTransaction.java
+++ b/src/main/java/com/stripe/model/BalanceTransaction.java
@@ -23,12 +23,10 @@ public class BalanceTransaction extends APIResource implements HasId {
 	Long fee;
 	List<Fee> feeDetails;
 	Integer net;
-	String source;
+	ExpandableField<HasId> source;
 	String status;
 	String type;
 	
-	transient HasId sourceObject;
-	
 	@Deprecated
 	TransferCollection sourcedTransfers;
 
@@ -112,14 +110,6 @@ public class BalanceTransaction extends APIResource implements HasId {
 		this.net = net;
 	}
 
-	public String getSource() {
-		return source;
-	}
-
-	public void setSource(String source) {
-		this.source = source;
-	}
-
 	/**
 	 * @deprecated
 	 * Recent API versions no longer return this field (https://stripe.com/docs/upgrades#2017-01-27).
@@ -148,14 +138,34 @@ public class BalanceTransaction extends APIResource implements HasId {
 	public void setType(String type) {
 		this.type = type;
 	}
-	
-	public void setSourceObject(HasId sourceObject) {
-		this.sourceObject = sourceObject;
+
+	public String getSource() {
+		if (this.source == null) {
+			return null;
+		}
+		return this.source.getId();
+	}
+
+	public void setSource(String sourceID) {
+		this.source = setExpandableFieldID(sourceID, this.source);
 	}
 
-	@SuppressWarnings("unchecked")
-	public <O> O getSourceObject() {
-		return (O) sourceObject;
+	public HasId getSourceObject() {
+		if (this.source == null) {
+			return null;
+		}
+		return this.source.getExpanded();
+	}
+
+	public void setSourceObject(HasId o) {
+		this.source = new ExpandableField<HasId>(o.getId(), o);
+	}
+
+	public <O extends HasId> O getSourceObjectAs() {
+		if (this.source == null) {
+			return null;
+		}
+		return (O) this.source.getExpanded();
 	}
 		
 	public static BalanceTransaction retrieve(String id) throws AuthenticationException,
```

Naming the generic accessor `getSourceObjectAs` gives us a plausible avenue to change `EventData` to have a similar method without having to introduce a backwards incompatible change.